apps/storybook/src/overlays/Menu.stories.tsx

```diff
@@ -0,0 +1,63 @@
+import type { Meta, StoryObj } from "@storybook/react";
+
+import {
+  Button,
+  Menu,
+  MenuContent,
+  MenuItem,
+  MenuTrigger,
+} from "@optiaxiom/react";
+import { expect, screen, userEvent, within } from "@storybook/test";
+import { IconStar } from "@tabler/icons-react";
+
+const meta: Meta<typeof Menu> = {
+  component: Menu,
+};
+
+export default meta;
+
+type Story = StoryObj<typeof Menu>;
+
+export const Basic: Story = {
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+
+    const buttons = canvas.getAllByRole("button", {
+      name: "Trigger Menu",
+    });
+    const button = buttons[0];
```

Comment:
I think this catches a design issue regarding the trigger. We are wrapping button within a button. As `MenuTrigger` itself is a button, we are passing another `button` for it.
We can use composition in `MenuTrigger` component. So, the child will be the only button. And we wont need to handle it like this rather just 

```suggestion
    const button = canvas.getByRole("button", {
      name: "Trigger Menu",
    });
```